### PR TITLE
MNT: setup: Specify mock as dependency

### PR DESCRIPTION
**Pull request by kyleam**
mock is assumed to be present because it's a dependency of datalad.
However the master branch of datalad no longer supports Python 2, so
it's free to drop its dependency on the external mock library in favor
of unittest.mock, which comes with the standard library as of Python
3.3.

Explicitly specify mock as a requirement for the tests.

Re: https://github.com/datalad/datalad/pull/3860

### CODE REVIEW

**Comment by yarikoptic** [comment]
Yeap: https://github.com/datalad/datalad-container/pull/110/files

**Comment by kyleam** [comment]
Yes, you're right that this should just follow what datalad and datalad-container do (and it would've needed additional tweaking to work).  I've updated it. 

Thanks.

**Comment by yarikoptic** [comment]
What we ultimately want is to use datalad_setup in all extensions. Main one uses it already https://github.com/datalad/datalad/blob/master/setup.py#L178 but I didn't get to change extensions yet, or may be adjust it first to make things even more specific

**Comment by kyleam** [comment]
> What we ultimately want is to use datalad_setup in all extensions.

Yes, I reviewed that PR in datalad core, and I understand that's what you eventually want.  I'm not clear if you're asking for me to incorporate that here, but if so: no, I'd prefer not to.

File: .travis.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@ script:
   # -A "$NOSE_SELECTION_OP(integration or usecase or slow)"
   - http_proxy=
     PATH=$PWD/tools/coverage-bin:$PATH
-    $NOSE_WRAPPER `which nosetests` $NOSE_OPTS
+    $NOSE_WRAPPER python -m $NOSE_OPTS
```
Comment (yarikoptic) [inline .travis.yml]: Here needs -m nose before options

**Pull request merged**